mm2src/coins/eth.rs

```diff
@@ -3240,6 +3353,56 @@ impl EthCoin {
         };
         Box::new(fut.boxed().compat())
     }
+
+    /// Checks every second till ETH nodes recognize that nonce is increased.
+    /// Parity has reliable "nextNonce" method that always returns correct nonce for address.
+    /// But we can't expect that all nodes will always be Parity.
+    /// Some of ETH forks use Geth only so they don't have Parity nodes at all.
+    ///
+    /// Please note that we just keep looping in case of a transport error hoping it will go away.
+    ///
+    /// # Warning
+    ///
+    /// The function is endless, we just keep looping in case of a transport error hoping it will go away.
+    async fn wait_for_addr_nonce_increase(&self, addr: Address, prev_nonce: U256) {
+        repeatable!(async {
```

Comment:
Yes, correct)
>  I guess it was implemented when I was in the military :)

No, because `wait_for_addr_nonce_increase` uses `until_ready`. If @artemii235 confirms that we can use a timeout here https://github.com/KomodoPlatform/atomicDEX-API/pull/1591#discussion_r1064496216, I'll change the function to use `Repeatable::with_timeout` or `Repeatable::attempts` restrictions.
> Q: Will this process kill itself somehow if the fut never gets ready?

Reply:
This is a huge problem specially when processes stacks with multiple other processes. Imagine having ~10 or even more processes running infinitly. Is it worth to create an issue for this to make sure we don't miss this problem? @artemii235

Reply:
I believe this can be resolved as we discussed during the meeting

Reply:
Correct, marked as resolved